lib/scope.js

```diff
@@ -168,7 +168,8 @@ class SymbolDef {
                   || this.orig[0] instanceof AST_SymbolDefun) && keep_name(options.keep_fnames, this.orig[0].name)
             || this.orig[0] instanceof AST_SymbolMethod
             || (this.orig[0] instanceof AST_SymbolClass
-                  || this.orig[0] instanceof AST_SymbolDefClass) && keep_name(options.keep_classnames, this.orig[0].name);
+                  || this.orig[0] instanceof AST_SymbolDefClass) && keep_name(options.keep_classnames, this.orig[0].name)
+            || (this.orig[0] instanceof AST_SymbolImport && options.keep_imports);
```

Comment:
As per your comment, maybe what's needed to solve the problem is a heuristic such as keeping imports that are used less than 2 or 3 times. `this.references.length` should contain this information.